apps/core/components/ItineraryCard/ItineraryCard.js

```diff
@@ -0,0 +1,112 @@
+// @flow
+
+import * as React from 'react';
+import { View, FlatList } from 'react-native';
+import shortid from 'shortid';
+import { LocalizedPrice, StyleSheet } from '@kiwicom/universal-components';
+import { formatPrice } from '@kiwicom/margarita-utils';
+import { defaultTokens } from '@kiwicom/orbit-design-tokens';
+import { graphql, createFragmentContainer } from '@kiwicom/margarita-relay';
+
+import ItineraryCardRow from './ItineraryCardRow';
+import BadgesContainer from './BadgesContainer';
+import HorizontalDash from './HorizontalDash';
+import TripSector from './TripSector';
+import type { TripSector as TripSectorType } from './__generated__/TripSector.graphql';
+import type { ItineraryCard as ItineraryCardType } from './__generated__/ItineraryCard.graphql';
+
+type Props = {|
+  +data: ItineraryCardType,
+|};
+
+type SectorItem = {|
+  item: ?{
+    ...TripSectorType,
+    +$fragmentRefs: any,
+  },
+|};
+
+class ItineraryCard extends React.Component<Props> {
+  renderSectorItem = ({ item }: SectorItem) => {
+    if (item) {
+      return (
+        <ItineraryCardRow>
+          <TripSector data={item} />
+        </ItineraryCardRow>
+      );
+    }
+    return null;
+  };
+
+  keyExtractor = () => shortid.generate();
```

Comment:
Please use the id field for this purpose.

Reply:
But server is not getting you any ID of sector

Reply:
Then we should create an ID for sector

Reply:
You should use stable keys for this purpose. It's not complicated to add a new unique and stable ID. You could as well replace this with `keyExtractor = (_, index) => index;` which does the same thing but without any external library.

Reply:
So is it OK generate ID on the server if there is no exist ID? 
I think that it can lead to confusion because it creates the expectation that you can ask for a particular sector by ID, but it is no truth.

Reply:
Should be fine, yes. ID doesn't say anything about fetching. It just uniquely represents the type (exactly what you need here). Later you could use it for fetching without BC breaks if there is going to be such a query.

As I said - if you don't want it's fine to use just `keyExtractor = (_, index) => index` - it's better than generating random keys every time. Keep it simple.

Reply:
@mrtnzlml 
thanks for the explanation, 👍 make it more sense to me now.